framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts

```diff
@@ -39,3 +41,26 @@ export const configSchema = {
 		livenessFrequency: 86400,
 	},
 };
+
+export const chainConnectorInfoSchema = {
+	$id: '#/plugins/chainConnector/info',
+	type: 'object',
+	properties: {
+		blockHeaders: {
+			type: 'array',
+			fieldNumber: 1,
+			items: {
+				...chain.blockHeaderSchema,
+			},
+		},
+		aggregateCommits: {
+			type: 'object',
+			fieldNumber: 2,
+		},
+		validatorsHashPreimage: {
+			type: 'object',
+			fieldNumber: 3,
+		},
```

Comment:
I think we need to expand these

Reply:
```	aggregateCommits: Record<number, AggregateCommit>;
	validatorsHashPreimage: Record<string, ValidatorsData>;
```
Can you maybe give an example how to expand them ?

Reply:
as we cannot use (key, value) pair in codec unless we create a substore. we can use array of [aggregateCommit Schema](https://github.com/LiskHQ/lisk-sdk/blob/e98d0ecd7aa67cbbd3ed35199c9ea99f7d31d594/framework/src/engine/consensus/certificate_generation/schema.ts#L96)

validatorsHashPreimage we can use array of 

```ts
validatorsData {
        prevoteThreshold: bigint;
	precommitThreshold: bigint;
	certificateThreshold: bigint;
	validators: BFTValidator[];
	validatorsHash: Buffer;
} []
```
which we are getting from `getBFTParameters` function in BFT

Reply:
Thank you, can you maybe also formulate a example in terms of exact schema for atleast one of them like
```
	aggregateCommits: {
			type: 'object',

			fieldNumber: 2,
		},
		validatorsHashPreimage: {
			type: 'object',
			fieldNumber: 3,
		},
```